### ISSUE TITLE
[Bug]: aws_ami Data Source Not Returning Results

### ISSUE THREAD

**Issue by chakatz** [hashicorp/terraform-provider-aws#37709]
### Terraform Core Version

1.7.5,1.8.4

### AWS Provider Version

5.51.0,5.51.1

### Affected Resource(s)

aws_ami data source

### Expected Behavior

An AMI that meets our filter criteria should be returned.

### Actual Behavior

> Error: Your query returned no results. Please change your search criteria and try again.

### Terraform Configuration Files

```
data "aws_ami" "wordpress-ami" {
  most_recent = true
  filter {
    name   = "name"
    values = ["Wordpress-AMI-*"]
  }
  filter {
    name   = "tag:approved-ami"
    values = [""]
  }
}
```

### Steps to Reproduce

This same code works with AWS providers before `5.51.0` and fails with `5.51.0` and `5.51.1`

### Would you like to implement a fix?

None

**Comment by YakDriver** [comment]
Is it possible the misspelling in `values` is causing the problem?

```
"Worpress-AMI-*
"Wordpress-AMI-*"
```

**Comment by chakatz** [comment]
> Is it possible the misspelling in `values` is causing the problem?
> 
> ```
> "Worpress-AMI-*
> "Wordpress-AMI-*"
> ```

Hi @justinretzolk,
Good eye. No, that's not it. I changed our actual code here in the ticket and typed it wrong. I've fixed it in my comment.
It's failing on all the ami searched in our code, not just this one. When I changed the provider to 5.50.0, it works.

**Comment by nam054** [comment]
It seems the issue was a result from a bug fix in `aws_sdk_go` where they no longer serialize empty non-nil lists. The fix will be released in 5.54.0.
